src/components/UserSelector.tsx

```diff
@@ -20,21 +40,22 @@
 
       <div className="dropdown-menu" id="dropdown-menu" role="menu">
         <div className="dropdown-content">
-          <a href="#user-1" className="dropdown-item">
-            Leanne Graham
-          </a>
-          <a href="#user-2" className="dropdown-item is-active">
-            Ervin Howell
-          </a>
-          <a href="#user-3" className="dropdown-item">
-            Clementine Bauch
-          </a>
-          <a href="#user-4" className="dropdown-item">
-            Patricia Lebsack
-          </a>
-          <a href="#user-5" className="dropdown-item">
-            Chelsey Dietrich
-          </a>
+          {users.map(user => {
+            const { id, name } = user;
+
+            const isSelected = id === selectedUser?.id;
+
+            return (
+              <a
+                href={`#user-${id}`}
+                className={cn('dropdown-item', { 'is-active': isSelected })}
+                key={id}
+                onMouseDown={() => onSelectUser(user)}
```

Comment:
Using `onMouseDown` to select a user might lead to unexpected behavior, such as triggering the selection before the click is completed. Consider using `onClick` instead to ensure the action occurs after the click event.